LevelComponents/Level.cpp

```diff
@@ -9,77 +9,133 @@
 
 extern WL4EditorWindow *singleton;
 
-/// <summary>
-/// Helper function to create a level name string from a data address in the ROM.
-/// </summary>
-/// <param name="address">
-/// Starting address of the level name string.
-/// </param>
-/// <returns>
-/// The string, as a QString
-/// </returns>
-static QString ConvertDataToLevelName(int address)
+namespace LevelComponents
 {
-    QString ret = "";
-    for (int i = 0; i < 26; i++)
+    /// <summary>
+    /// Helper function to Generate LevelName available character QString.
+    /// </summary>
+    QString Level::GetAvailableLevelNameChars()
     {
-        unsigned char chr = ROMUtils::CurrentFile[address + i];
-        if (chr <= 0x09)
-        {
-            ret += chr + 48;
-        }
-        else if (chr >= 0x0A && chr <= 0x23)
-        {
-            ret += chr + 55;
-        }
-        else if (chr >= 0x24 && chr <= 0x3D)
-        {
-            ret += chr + 61;
-        }
-        else
-        {
-            ret += (unsigned char) 32;
-        }
+        /*"0123456789ABCDEFGHIJKLMNOPQRSTUV"
+        "WXYZabcdefghijklmnopqrstuvwxyz.&"
+        "あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほまみ"
+        "むめもやゆよらりるれろわをんぁぃぅぇぉゃゅょっがぎぐげござじずぜ"
+        "ぞだぢづでどばびぶべぼぱぴぷぺぽアイウエオカキクケコサシスセソタ"
+        "チツテトナニヌネノハヒフヘホマミムメモヤユヨラリルレロワヲンァィ"
+        "ゥェォャュョッガギグゲゴザジズゼゾダヂヅデドバビブベボパピプペポ"
+        "ヴ'、。—~…!?()「」『』[]℃-"*/
+        unsigned char tmpstr[484] = "\xe3\x81\x82\xe3\x81\x84\xe3\x81\x86\xe3\x81\x88\xe3\x81\x8a"
```

Comment:
instead of `unsigned char tmpstr[484]` make this `const unsigned char *tmpstr` (and the other strings which will not change)

Reply:
Actually just do const char*. That way you do not need to cast below. The main point is these should not be arrays.

Reply:
Done.